plugins/@grouparoo/app-templates/src/source/shared/recordProperty.ts

```diff
@@ -33,13 +36,14 @@ export const getRecordProperty: GetRecordPropertyMethod = ({
     propertyOptions,
     propertyFilters,
   }) => {
+    const sourceQuery = sourceOptions[sourceQueryKey]?.toString();
     const tableName = sourceOptions[tableNameKey]?.toString();
     const matchName = Object.keys(sourceMapping)[0]; // tableCol
     const recordPropertyMatch = Object.values(sourceMapping)[0];
     const columnName = propertyOptions[columnNameKey]?.toString();
-    const aggregationMethod = <AggregationMethod>(
-      propertyOptions[aggregationMethodKey]
-    );
+    const aggregationMethod = useAggregations
+      ? <AggregationMethod>propertyOptions[aggregationMethodKey]
+      : AggregationMethod.Exact;
```

Comment:
Query sources don't have an `aggregationMethod` option, but we still want to group all the properties so we can get them in one go. Because of this I'm now defaulting to `exact` in core when determining groupings. Because the plugin is still the one saying which aggregations can be grouped via `groupAggregations`, I think this is safe.

Initially I had the aggregation method option for query sources too, just with a single option to select, but this added an extra unnecessary step in the UI that I don't think should be there only for internal reasons.